Framework/DataObjects/inc/MantidDataObjects/TableColumn.h

```diff
@@ -162,7 +162,17 @@ template <class Type> class TableColumn : public API::Column {
         typename boost::mpl::if_c<boost::is_convertible<double, Type>::value,
                                   Type, InconvertibleToDoubleType>::type
             DoubleType;
-    return boost::numeric_cast<double, DoubleType>(m_data[i]);
+    if (typeid(m_data[i]) == typeid(std::string)) {
```

Comment:
I am wondering if we should have something like
```cpp
  auto isStringType = std::is_convertible<Type, std::string>::value;
  if (isStringType) {
     ...
```
 here and avoid the `typeid`. To be fair it is probably a compile-time operation anyway in this case. I just don't come across the need to use `typeid` very often. @martyngigg  do you have an opinion on this?